### PR TITLE
feat: allow to provide cookies to authenticate to Nuvla 

### PR DESCRIPTION
**Pull request by ignacio-penas**
Closes sixsq/tasklist#3225

### CODE REVIEW

File: nuvla/job_engine/job/base.py [hunk 1]
```diff
@@ -148,14 +152,34 @@ def _init_statsd(self):
...
+            os.mkdir(cookie_dir, stat.S_IRUSR | stat.S_IWUSR | stat.S_IXUSR)
+
+        with open(cookie_file, "w") as f:
+            f.write(decoded_cookies)
+
     def _init_nuvla_api(self):
         # true unless header authentication is used
-        reauthenticate = self.args.api_authn_header is None
+        cookies = os.getenv("JOB_COOKIES", "")
+        persist_cookie = cookies != "" and cookies is not None
```
Comment (schaubl) [inline nuvla/job_engine/job/base.py:168]: could be simplified with
```suggestion
        persist_cookie = bool(cookies)
```

**Review by schaubl — COMMENTED** [review]
I would use a tmpfile for the cookie file.

But in fact, why writing the cookie to a file ?

After the Api instance is created it's possible to do:
```python
self.api = Api(...)
self.api.session.cookies.set_cookie(cookie)
```
In this case it needs to be a (single) cookie like returned by the Set-Cookie HTTP header.
But you can load a mozilla cookie file in memory with `BytesIO` (or `StringIO`):
```python
self.api = Api(...)
self.api.session.cookies = MozillaCookieJar()
self.api.session.cookies._really_load(BytesIO(cookies), '', ignore_discard=True)
```

**Comment by schaubl** [comment]
Please also see the review comment in the other PR: https://github.com/nuvlaedge/nuvlaedge/pull/214#pullrequestreview-2416364011

**Comment by ignacio-penas** [comment]
> I would use a tmpfile for the cookie file.
> 
> But in fact, why writing the cookie to a file ?
> 
> After the Api instance is created it's possible to do:
> 
> ```python
> self.api = Api(...)
> self.api.session.cookies.set_cookie(cookie)
> ```
> 
> In this case it needs to be a (single) cookie like returned by the Set-Cookie HTTP header. But you can load a mozilla cookie file in memory with `BytesIO` (or `StringIO`):
> 
> ```python
> self.api = Api(...)
> self.api.session.cookies = MozillaCookieJar()
> self.api.session.cookies._really_load(BytesIO(cookies), '', ignore_discard=True)
> ```

Doing it this way because cookies are loaded on the constructor of SessionStore from the file parsed as a parameter (or default) when persist_cookie) is present. And, to be fair, I tried setting the cookies with your method and fail. Mainly because MozillaCookieJar do not have a get/retrieve method to extract a serialised version of the cookies.

**Pull request merged**